### ISSUE TITLE
[Performance] Refresh Access Token Failed with Status code: 400

### ISSUE THREAD

**Issue by sharmadine** [Microsoft/BotBuilder#3196]
## System Information (Required)
* SDK Language: NODE
* SDK Version: 3.7.0
* Development Environment: LOCALHOST, AZURE_APP_SERVICE

## Issue Description
Our Production Bot is deployed and configured for FB, Slack, Skype, Web. It was working perfectly well across channels. There's no new code / release to the prod env. 

Since 30 mins, the bot is down with errors in log files - refresh access token issue.
Please find the log entries below.

I am not sure what's the issue. The Bot just stopped working, which seems some issues with Token Issue / Refresh API Endpoints. 
```
2017-07-25T18:25:04.264999000Z Error: Refresh access token failed with status code: 400
2017-07-25T18:25:04.265192000Z     at Request._callback (/usr/src/app/node_modules/botbuilder/lib/bots/ChatConnector.js:489:28)
2017-07-25T18:25:04.265351000Z     at Request.self.callback (/usr/src/app/node_modules/request/request.js:187:22)
2017-07-25T18:25:04.265478000Z     at emitTwo (events.js:106:13)
2017-07-25T18:25:04.265627000Z     at Request.emit (events.js:194:7)
2017-07-25T18:25:04.265760000Z     at Request.<anonymous> (/usr/src/app/node_modules/request/request.js:1044:10)
2017-07-25T18:25:04.265870000Z     at emitOne (events.js:96:13)
2017-07-25T18:25:04.266006000Z     at Request.emit (events.js:191:7)
2017-07-25T18:25:04.266130000Z     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/request/request.js:965:12)
2017-07-25T18:25:04.266257000Z     at emitNone (events.js:91:20)
2017-07-25T18:25:04.266393000Z     at IncomingMessage.emit (events.js:188:7)
2017-07-25T18:25:04.266521000Z     at endReadableNT (_stream_readable.js:975:12)
2017-07-25T18:25:04.266649000Z     at _combinedTickCallback (internal/process/next_tick.js:80:11)
2017-07-25T18:25:04.266826000Z     at process._tickDomainCallback (internal/process/next_tick.js:128:9)2017-07-25T18:25:04.264999000Z Error: Refresh access token failed with status code: 400
2017-07-25T18:25:04.265192000Z     at Request._callback (/usr/src/app/node_modules/botbuilder/lib/bots/ChatConnector.js:489:28)
2017-07-25T18:25:04.265351000Z     at Request.self.callback (/usr/src/app/node_modules/request/request.js:187:22)
2017-07-25T18:25:04.265478000Z     at emitTwo (events.js:106:13)
2017-07-25T18:25:04.265627000Z     at Request.emit (events.js:194:7)
2017-07-25T18:25:04.265760000Z     at Request.<anonymous> (/usr/src/app/node_modules/request/request.js:1044:10)
2017-07-25T18:25:04.265870000Z     at emitOne (events.js:96:13)
2017-07-25T18:25:04.266006000Z     at Request.emit (events.js:191:7)
2017-07-25T18:25:04.266130000Z     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/request/request.js:965:12)
2017-07-25T18:25:04.266257000Z     at emitNone (events.js:91:20)
2017-07-25T18:25:04.266393000Z     at IncomingMessage.emit (events.js:188:7)
2017-07-25T18:25:04.266521000Z     at endReadableNT (_stream_readable.js:975:12)
2017-07-25T18:25:04.266649000Z     at _combinedTickCallback (internal/process/next_tick.js:80:11)
2017-07-25T18:25:04.266826000Z     at process._tickDomainCallback (internal/process/next_tick.js:128:9)
```

**Comment by nwhitmont** [comment]
@sharmadine Thanks for reporting. We are experiencing a service interruption and we've got multiple engineers looking into it. Will update you when we have more details.

**Comment by Cartman720** [comment]
Hi @nwhitmont 

Same here
Thank you for fast response.
We will wait

**Comment by rrdelaney** [comment]
Also having the issue.

**Comment by srozga** [comment]
+1, for myself and my clients.

**Comment by toxtli** [comment]
+1 Let us know when it is fixed

**Comment by mdeshields** [comment]
+1, arg

**Comment by rmimura** [comment]
+1 bot is on mute

**Comment by ericvanisko** [comment]
+1

**Comment by stevefox** [comment]
+1

**Comment by luis-sambiase** [comment]
+1

**Comment by xjose97x** [comment]
+1

**Comment by rcarmonad** [comment]
plus one

**Comment by LeonBirk** [comment]
same here

**Comment by McBacker** [comment]
We all know that it is still in Preview, but, for those who have their Bots under BotFramework infraestructure... how do you explain this downtime to your customers?

**Comment by binhpht** [comment]
same

**Comment by mpuchi** [comment]
+1

**Comment by yavuzKomecoglu** [comment]
I seem to have improved

**Comment by toxtli** [comment]
The service is restored :)

**Comment by nwhitmont** [comment]
The incident has been resolved, bots should be back to normal activity now.

**Comment by sharmadine** [comment]
Yes, the bot is alive again... thanks.

**Comment by xjose97x** [comment]
my bot is still not replying?

**Comment by xjose97x** [comment]
Notice last message says: **Error sending to bot**

This issue happens on all channels except emulator.
I'm even remote debugging to my azure app service, and bot framework is not sending the activities to it.
![Image of errors](https://preview.ibb.co/ej42Uk/Captura.png)

**Comment by xjose97x** [comment]
I apologize, I just found the issue. It made me get confused since the issue appeared along this one and was impossible for me to debug.

So, I changed the namespace and project name of my bot application. Apparently this made Azure app service believe there were multiple API controller endpoints somehow. And that's why the Bot Framework connector couldn't reach my bot application.
So even after cleaning my solution, recompiling, deleting bin and obj folders, etc it didn't work. What I had to do is delete the azure app service and create a new one, and voila!

Sorry for the inconveniences!
